docs/config.rst

```diff
@@ -0,0 +1,134 @@
+.. -*- coding: utf-8 -*-
+
+.. _config:
+
+Configuration Storage
+=====================
+
+Context and History
+-------------------
+
+`magic-folder` originally was extracted from Tahoe-LAFs where it was a
+sub-command / sub-system. Thus, it inherited Tahoe's method of
+configuration. This is a menagerie of various files in a "node
+directory" including a `tahoe.cfg` INI-style file. There is a
+`private/` sub-directory which often has "extra-private" information
+(like secret keys).
+
+
+Although a magic-folder is necessarily tied to a Tahoe-LAFS client we
+don't *need* to store our configuration there.
+
+We have decided that a user upgrading from a `tahoe magic-folder`
+style Magic Folder to a stand-alone Magic Folder will run
+`magic-folder migrate`. This allows us the freedom to store our
+configuration in any way we like.
+
+
+Legacy Configuration and State
+------------------------------
+
+In versions of Tahoe-LAFS that had magic-folder enabled, the following
+configuration could be used:
+
+- `enabled` (bool) from `tahoe.cfg [magic_folder]`
+- `local.directory` (from `tahoe.cfg [magic_folder]`)
+- `poll_interval` (from `tahoe.cfg [magic_folder]`)
+
+- "magic folders" dict from
+  `<node_dir>/private/magic_folders.yaml`. This uses magic-folder
+  "names" as keys with the following valid keys as sub-dicts:
+  - `directory`: local path to the "magic folder"
+  - `poll_interval`: how often to poll (in seconds)
+  - `collective_dircap`: read-cap of the shared directory (list of users)
+  - `upload_dircap`: write-cap of our mutable directory
+  - `umask`: the umask to use for downloaded files
+
+- `<node_dir>/private/magicfolder_<name>.sqlite` the local sqlite
+  database for tracking state
+
+Related, Tahoe-LAFS has a token in `<node_dir>/private/api_auth_token`
+which is required to access parts of the Web API. This isn't directly
+part of magic-folder's configuration (however, standalone magic-folder
+will also use such a token).
+
+
+What Needs to be Stored
+-----------------------
+
+As we are currently re-designing the storage mechanism, some of this
+might change. However, based on current understanding the state we
+WANT to store is:
+
+- for each magic-folder (indexed by "name"):
+  - local directory path (the "magic folder")
+  - an author:
+    - name
+    - private_key (32 bytes, a NaCl signing key)
+  - a write-cap of our mutable directory
+  - some state (see below)
+- the node-directory of our Tahoe-LAFS client
+  - we may need the `private/api_auth_token`
+  - we will need the `node.url`
+  - (it MAY be sufficient to only store the API endpoint, not node-directory)
+- our own API endpoint (a Twisted server endpoint-string,
+  e.g. `tcp:1234` or `unix:/tmp/foo`)
+
+State that we need to store (per magic-folder):
+
+- a secret token that protects the API (only users who can read the
+  file can access the endpoint)
```

Comment:
I will re-word (my intention wasn't to change the token to be per-magic-folder, it just looks like that because of the "(per magic-folder)" parenthetical.